components/places/src/storage/history.rs

```diff
@@ -188,10 +188,15 @@ fn add_visit(
 
 /// Returns the GUID for the specified Url, or None if it doesn't exist.
 pub fn url_to_guid(db: &PlacesDb, url: &Url) -> Result<Option<SyncGuid>> {
+    string_to_guid(db, url.clone().into_string())
+}
+
+/// Returns the GUID for the specified Url String, or None if it doesn't exist.
+pub fn string_to_guid(db: &PlacesDb, url: String) -> Result<Option<SyncGuid>> {
```

Comment:
Nit: This should probably take a `&str`, and `url_to_guid` should use `url.as_str()`, (It should have already been doing this, but nobody noticed it during the review, presumably).

Maybe rename it to `href_to_guid` too (we use `href` which we use in a few places to refer to a string containing a url).

Reply:
Done